aspnetcore/test/integration-tests/samples/3.x/IntegrationTestsSample/tests/RazorPagesProject.Tests/IntegrationTests/AuthTests.cs

```diff
@@ -0,0 +1,157 @@
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, profileWithUserName.StatusCode);
+            var profileWithUserHtml = 
+                await HtmlHelpers.GetDocumentAsync(profileWithUserName);
+            var userLogin = profileWithUserHtml.QuerySelector("#user-login");
+            Assert.Equal("user", userLogin.TextContent);
+        }
+
+        public class TestGithubClient : IGithubClient
+        {
+            public Task<GithubUser> GetUserAsync(string userName)
+            {
+                if (userName == "user")
+                {
+                    return Task.FromResult(
+                        new GithubUser
+                        {
+                            Login = "user",
+                            Company = "Contoso Blockchain",
+                            Name = "John Doe"
+                        });
+                }
+                else
+                {
+                    return Task.FromResult<GithubUser>(null);
+                }
+            }
+        }
+        #endregion
+
+        #region snippet2
+        [Fact]
+        public async Task Get_SecurePageRedirectsAnUnauthenticatedUser()
+        {
+            // Arrange
+            var client = _factory.CreateClient(
+                new WebApplicationFactoryClientOptions
+                {
+                    AllowAutoRedirect = false
+                });
+
+            // Act
+            var response = await client.GetAsync("/SecurePage");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+            Assert.StartsWith("http://localhost/Identity/Account/Login", 
+                response.Headers.Location.OriginalString);
+        }
+        #endregion
+
+        #region snippet3
+        [Fact]
+        public async Task Get_SecurePageIsReturnedForAnAuthenticatedUser()
+        {
+            // Arrange
+            var client = _factory.WithWebHostBuilder(builder =>
+                {
+                    builder.ConfigureTestServices(services =>
+                    {
+                        services.AddAuthentication("Test")
+                            .AddScheme<AuthenticationSchemeOptions, AuthTestHandler>(
+                                "Test", options => {});
+                    });
+                })
+                .CreateClient(new WebApplicationFactoryClientOptions
+                {
+                    AllowAutoRedirect = false,
+                });
+
+            client.DefaultRequestHeaders.Authorization = 
+                new AuthenticationHeaderValue("Test");
+
+            //Act
+            var response = await client.GetAsync("/SecurePage");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+        #endregion
+    }
+
+    #region snippet4
+    public class AuthTestHandler : AuthenticationHandler<AuthenticationSchemeOptions>
```

Comment:
Consider the name "TestAuthenticationHandler" since the name of the base class is `AuthenticationHandler`.

Reply:
ah ... I know why I did that now. It's line length ... we have a fairly strict limit on 85 characters to prevent horizontal scrollbars. No worries. I'll break them differently to allow the topic+sample to use it.

Reply:
🤔 Let's go with `TestAuthHandler` ... it just doesn't compose well being so long for our line length rule.